include/mips-emulator/executor.hpp

```diff
@@ -1,8 +1,19 @@
 #pragma once
 #include "mips-emulator/instruction.hpp"
+#include "memory.hpp"
+#include "register_file.hpp"
 
 namespace mips_emulator {
     namespace Executor {
+
+        // Declare non templated functions
```

Comment:
You don't need to declare them if the implementation is in the header file already.